portal-web/docroot/html/taglib/aui/nav_item/end.jsp

```diff
@@ -31,10 +31,10 @@ if (Validator.isNull(title)) {
 %>
 
 <c:if test="<%= !dropdown || Validator.isNotNull(bodyContentString.trim()) %>">
-	<li class="<%= cssClass %><%= selected ? " active " : StringPool.SPACE %><%= state %>" id="<%= id %>" role="presentation" <%= AUIUtil.buildData(data) %> <%= InlineUtil.buildDynamicAttributes(dynamicAttributes) %>>
+	<li class="<%= !dropdown ? " nav-item " : ""%><%= cssClass %><%= selected ? " active " : StringPool.SPACE %><%= state %>" id="<%= id %>" role="presentation" <%= AUIUtil.buildData(data) %> <%= InlineUtil.buildDynamicAttributes(dynamicAttributes) %>>
```

Comment:
Still not a big fan of adding these classes here even with the dropdown check. We've had it without any default classes for so long that it feels wrong to add it now.

Reply:
I know but the other solution implies that we need to add the class every time we use `<aui:nav-item>`, which doesn't seem big in the portal but it could be the wrong pattern (shouldn't we put the bigger part of code in the taglib, in order to avoid repetition?)

Reply:
Unfortunately there is no clear "right" or "wrong" here:

- Adding to the taglib reduces duplication 🎉  but changes default behavior in a way that may surprise users 😢
- Adding classes everywhere avoids the downside of the behavior change (because there is no behavior change) 🎉  but is error prone (because we might miss a spot) 😢

I would leave this open for a day to give people time to comment on the trade-off, but if there's no clear consensus then we just have to make a call and live with the consequences. (I personally, don't feel strongly one way or the other, although I like the idea of centralizing the fix.)

Reply:
To add more context, before we had the compat-layer underneath that worked like the taglib is doing now, so from a user perspective it should be "invisible" as always. But yeah, I share your doubts and since we hopefully will get rid of the taglib both solutions are fine to me

Reply:
@edalgrin you're right about compat layer I completely forgot about that. I think we can disregard this.